src/components/icon/icon.pw.tsx

```diff
@@ -0,0 +1,538 @@
+      await expect(iconLocator).toHaveId(id);
+    });
+  });
+
+  testData.forEach((tooltipMessage) => {
+    test(`should check tooltipMessage as ${tooltipMessage}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<IconTooltipComponent tooltipMessage={tooltipMessage} />);
+
+      const iconLocator = await page.getByTestId("icon");
+      await iconLocator.hover();
+      const tooltip = getDataElementByValue(page, "tooltip");
+      await expect(tooltip).toHaveText(tooltipMessage);
+    });
+  });
+
+  (["top", "bottom", "left", "right"] as const).forEach((tooltipPosition) => {
+    test(`should check tooltip position as ${tooltipPosition}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<IconTooltipComponent tooltipPosition={tooltipPosition} />);
+
+      const tooltip = getDataElementByValue(page, "tooltip");
+      await expect(tooltip).toHaveAttribute("data-placement", tooltipPosition);
+    });
+  });
+
+  test("should check when tooltipVisible prop is set as true", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<IconTooltipComponent tooltipVisible />);
+    const tooltip = getDataElementByValue(page, "tooltip");
+    await expect(tooltip).toBeVisible();
+  });
+
+  test("should check when tooltipVisible prop is set as false", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<IconTooltipComponent tooltipVisible={false} />);
+    const tooltip = getDataElementByValue(page, "tooltip");
+    await expect(tooltip).not.toBeVisible();
+  });
+
+  colorData.forEach((tooltipBgColor) => {
+    test(`should check tooltip background color as ${tooltipBgColor}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<IconTooltipComponent tooltipBgColor={tooltipBgColor} />);
+
+      const tooltip = getDataElementByValue(page, "tooltip");
+      await expect(tooltip).toHaveCSS("background-color", tooltipBgColor);
+    });
+  });
+
+  colorData.forEach((tooltipFontColor) => {
+    test(`should check tooltip font color as ${tooltipFontColor}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<IconTooltipComponent tooltipFontColor={tooltipFontColor} />);
+
+      const tooltip = getDataElementByValue(page, "tooltip");
+      await expect(tooltip).toHaveCSS("color", tooltipFontColor);
+    });
+  });
+
+  testData.forEach((tooltipId) => {
+    test(`should check tooltipId as ${tooltipId}`, async ({ mount, page }) => {
+      await mount(<IconTooltipComponent tooltipId={tooltipId} />);
+
+      const tooltip = getDataElementByValue(page, "tooltip");
+      await expect(tooltip).toHaveId(tooltipId);
+    });
+  });
+
+  (["error", "add", "admin", "alert"] as IconProps["type"][]).forEach(
+    (iconType) => {
+      test(`should check iconType as ${iconType}`, async ({ mount, page }) => {
+        await mount(<IconComponent type={iconType} />);
+
+        const iconLocator = await page.getByTestId("icon");
```

Comment:
await is not necessary here. This will work: 
`const iconLocator = page.getByTestId("icon");`